File: Java-BackEnd/JansiSnack_BackEnd/Api-commons/src/main/java/tech/wetech/api/model/Lifestyle.java
package tech.wetech.api.model;

import jakarta.persistence.*;

@Entity
@Table(name = "lifestyles")
public class Lifestyle {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @Column(length = 100, nullable = false, unique = true)
  private String name;

  @Column()
  private String imageUrl;


  public Lifestyle() {
  }

  public Long getId() {
    return id;
  }

  public void setId(Long id) {
    this.id = id;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getImageUrl() {
    return imageUrl;
  }

  public void setImageUrl(String imageUrl) {
    this.imageUrl = imageUrl;
  }

  public void addDessert(Dessert dessert) {
    dessert.setCategory(this.name);
  }
}

File: Kotlin-Android/JansiSnack_Front/app/src/main/java/com/example/jetsnack/model/Lifestyle.kt
package com.example.jetsnack.model;

import androidx.room.Entity

@Entity(tableName = "lifestyles")
class Lifestyle(
        id: Long,
        imageUrl: String,
        name: String,
        // 你可以在这里添加 Category 特有的属性
) : Kind(id, imageUrl, name) // 正确地调用 Kind 的构造器